hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/AvroConvertor.java

```diff
@@ -101,4 +107,18 @@ public GenericRecord fromProtoMessage(Message message) {
     initSchema();
     return ProtoConversionUtil.convertToAvro(schema, message);
   }
+
+  public GenericRecord withKafkaFieldsAppended(ConsumerRecord consumerRecord) {
+    initSchema();
+    GenericRecord record = (GenericRecord) consumerRecord.value();
+    GenericRecordBuilder recordBuilder = new GenericRecordBuilder(this.schema);
```

Comment:
this.schema may not have the offset right? 
so, how are we adding additional fields in L 118, 119, 129 when the GenericRecord schema does not have these fields?

Reply:
It always will when this method is called. I added a comment to say it's a requirement. The only place it is used is inside of an if (this.shouldAddOffsets) block, which means that the schemaprovider is wrapped with the kafka offset post processor.